src/responder/nss/nss_utils.c

```diff
@@ -36,3 +36,76 @@ nss_get_pwfield(struct nss_ctx *nctx,
 
     return nctx->pwfield;
 }
+
+errno_t get_extra_data(TALLOC_CTX *mem_ctx,
+                       struct sss_domain_info *domain,
+                       const char override_space,
+                       struct ldb_message *msg,
+                       struct sized_data *extra_data)
+{
+    const char *name;
+    char *short_name;
+    size_t len = 0;
+    size_t pos = 0;
+    uint8_t *data;
+    const char *flat_name;
+    const char *sid_str;
+
+    if (domain == NULL || domain->name == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing domain name.");
+        return EINVAL;
+    }
+
+    /* Use domain->name if domain->flat_name is undefined */
+    /* FIXME: or should it be better "" ? */
+    flat_name = domain->flat_name != NULL ? domain->flat_name : domain->name;
```

Comment:
remove "!= NULL" if you are using this kind of short expression you may optimize that out as well